### PR TITLE
EMF module and merge tests

### PR DESCRIPTION
**Pull request by obrix**
Add emf module (empty for now) with three tests for destructive merge, merging view merge and merge from CGMES loader.
EMF test written from powsybl tutorial to be done.

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: emf/src/test/java/com/powsybl/emf/IGMmergeTests.java [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    }
+
+    private static void exportStateVariable(Network network, CgmesExportContext context, Path file) {
+        try (OutputStream out = Files.newOutputStream(file)) {
+            XMLStreamWriter writer = XmlUtil.initializeWriter(true, "    ", out);
+            StateVariablesExport.write(network, writer, context);
+        } catch (IOException | XMLStreamException e) {
+            throw new RuntimeException(e);
+        }
+    }
```
Comment (miovd) [inline emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:236]: Those methods can be refactor by adding a `Runnable` in parameters

**Review by miovd — CHANGES_REQUESTED** [review]
Some comments about technical changes!
Regarding the functional scope, I think this is good for a first version, we can merge it for it to be in the next release and start a new PR improving it (particularly checking xnode and tielines).

File: emf/src/test/java/com/powsybl/emf/IGMmergeTests.java [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class IGMmergeTests {
+
+    private static FileSystem fs;
+
+    @BeforeAll
+    static void setUp() {
```
[the remark this answers is not in the document]
Comment (obrix) [reply emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:38]: In fact BeforeAll force the setUp function to be static void, thus filesystem too. Should be fixed by moving to BeforeEach, not really necessary IMHO but could avoid read/write errors.

File: emf/src/test/java/com/powsybl/emf/IGMmergeTests.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+                        "MicroGridTestConfiguration_BC_NL_SV_V2.xml",
+                        "MicroGridTestConfiguration_BC_NL_TP_V2.xml"),
+                CgmesConformity1Catalog.microGridBaseCaseBoundaries());
+        Network networkBENL = Network.read(mergedResourcesBENL.dataSource());
+
+        Set<String> branchIds = new HashSet<>();
+        Set<String> generatorsId = new HashSet<>();
+        Set<String> voltageLevelIds = new HashSet<>();
+
+        networkBENL.getBranches().forEach(b -> branchIds.add(b.getId()));
```
[the remark this answers is not in the document]
Comment (obrix) [reply emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:154]: Thanks.

File: emf/src/test/java/com/powsybl/emf/IGMmergeTests.java [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+    private static FileSystem fs;
+
+    @BeforeAll
+    static void setUp() {
+        fs = Jimfs.newFileSystem(Configuration.unix());
+    }
+
+    @Test
+    void igmsDestructiveMerge() throws IOException {
```
[the remark this answers is not in the document]
Comment (obrix) [reply emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:43]: Actually there is a mix between org.junit and org.junit.jupiter.api in powsybl entsoe, the public identifier is detected as a code smell in the later. Do you have any guideline for this project ?

[the remark this answers is not in the document]
Comment (miovd) [reply emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:43]: It is better to use junit5 (org.junit.jupiter.api), we have to get rid of org.junit (we just haven't had the time to do it yet)

[the remark this answers is not in the document]
Comment (obrix) [reply emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:50]: So no public modifier for Test annotated methods and test classes ?

File: emf/src/test/java/com/powsybl/emf/IGMmergeTests.java (hunk 1 above)
Comment (obrix) [reply emf/src/test/java/com/powsybl/emf/IGMmergeTests.java]: I did it with a consumer functionnal interface.

File: emf/src/test/java/com/powsybl/emf/IGMmergeTests.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (miovd) [reply emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:50]: Indeed

File: emf/src/test/java/com/powsybl/emf/IGMmergeTests.java [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+import java.io.OutputStream;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.Consumer;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class IGMmergeTests {
```
Comment (miovd) [inline emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:35]: Can you add your name as `@author` in the javadoc?

File: emf/src/test/java/com/powsybl/emf/IGMmergeTests.java [hunk 6]
```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
```
Comment (miovd) [inline emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:6]: Can you add the SPDX?

File: emf/src/test/java/com/powsybl/emf/IGMmergeTests.java [hunk 7]
```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
```
Comment (miovd) [inline emf/src/test/java/com/powsybl/emf/IGMmergeTests.java:2]: *2023

**Review by miovd — APPROVED** [review]

**Pull request merged**